src/scrapper/spiders/slot_spider.py

```diff
@@ -122,252 +123,53 @@ def func(self, error):
+        for schedule in schedule_data:
+            date_format = "%Y-%m-%dT%H:%M:%S"
+            start_time = datetime.strptime(schedule["start"], date_format)
+            end_time = datetime.strptime(schedule["end"], date_format)
 
-    def extractClassSchedule(self, response, cell, day, start_time, duration, course_unit_id):
-        lesson_type = cell.xpath(
-            'b/text()').extract_first().strip().replace('(', '', 1).replace(')', '', 1)
-        table = cell.xpath('table/tr')
-        location = table.xpath('td/a/text()').extract_first()
-        professor_link = table.xpath('td[@class="textod"]//a/@href').extract_first()
-        is_composed = 'composto_doc' in professor_link
-        professor_id = professor_link.split('=')[1]
+            if(int(schedule["id"]) in slot_ids):
+                continue
 
-        clazz = cell.xpath('span/a')
-        class_name = clazz.xpath('text()').extract_first()
-        class_url = clazz.xpath('@href').extract_first()
+            slot_ids.add(int(schedule["id"]))
 
-        # If true, this means the class is composed of more than one class
-        # And an additional request must be made to obtain all classes
-        if "hor_geral.composto_desc" in class_url:
-            return response.follow(
-                class_url,
-                dont_filter=True,
-                meta={
-                    'course_unit_id': course_unit_id, 
-                    'lesson_type': lesson_type, 
-                    'start_time': start_time, 
-                    'is_composed': is_composed,
-                    'professor_id': professor_id, 
-                    'location': location, 
-                    'day': day,
-                    'duration': duration
-                },
-                callback=self.extractComposedClasses
+            yield Class(
+                id=schedule["id"],
+                name=schedule["name"],
+                course_unit_id=re.search(r'uc/(\d+)/', self.api_url).group(1),
+                last_updated=datetime.now()
             )
 
-        class_id = get_class_id(course_unit_id, class_name)
-        if (class_id != None):
-            return Slot(
-                lesson_type=lesson_type,
-                day=day,
-                start_time=start_time,
-                duration=duration,
-                location=location,
-                is_composed=is_composed,
-                professor_id=professor_id,
-                class_id=class_id,
+            yield Slot(
+                id=schedule["id"],
+                lesson_type=schedule["typology"]["acronym"],
+                day=self.days[schedule["week_days"][0]],
+                start_time=start_time.hour + (start_time.minute / 60),
+                duration=(end_time - start_time).total_seconds() / 3600,
+                location=schedule["rooms"][0]["name"],
+                is_composed=len(schedule["persons"]) > 0,
```

Comment:
Isn't the composed classes by more than one classes? I think you mean schedule["classes"]

Reply:
Yes, you are right :sweat_smile: 

I will change it